cobalt/media/base/metrics_provider.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/media/base/metrics_provider.h"
+
+#include "base/logging.h"
+#include "base/metrics/histogram_functions.h"
+
+
+namespace cobalt {
+namespace media {
+
+using starboard::ScopedLock;
+
+
+MediaMetricsProvider::~MediaMetricsProvider() {
+  if (!IsInitialized()) return;
+  ReportPipelineUMA();
+}
+
+std::string MediaMetricsProvider::GetUMANameForAVStream(
```

Comment:
You say this is based on Chromium's media metrics provider? Can we not just use that one?

Reply:
No, the Chrome Media Metrics Provider is a mojo service (IPC service) -
https://source.chromium.org/chromium/chromium/src/+/main:media/mojo/services/media_metrics_provider.h